src/Event/SerializableEventTrait.php

```diff
@@ -14,7 +14,15 @@ trait SerializableEventTrait
      */
     public function serialize()
     {
-        return serialize(get_object_vars($this));
+        return serialize($this->__serialize());
+    }
+
+    /**
+     * @return array<string,mixed>
+     */
+    public function __serialize()
+    {
+        return get_object_vars($this);
```

Comment:
Sure thats the right one?

Reply:
Yeap. Missed the change for __unserialize.

Reply:
@dereuromark The issue is fixed if you want to have another look